corehq/apps/users/tasks.py

```diff
@@ -376,6 +361,31 @@ def process_reporting_metadata_staging():
         process_reporting_metadata_staging.delay()
 
 
+def _process_reporting_metadata_staging(chunk_size=100):
+    from corehq.apps.users.models import UserReportingMetadataStaging
+    for pk in UserReportingMetadataStaging.objects.values_list('pk', flat=True).order_by('pk')[:chunk_size]:
+        with transaction.atomic():
+            record = UserReportingMetadataStaging.objects.select_for_update(skip_locked=True).get(pk=pk)
```

Comment:
What does `skip_locked=True` do with `.get(pk=pk)`? Does it raise `DoesNotExist`? I didn't see a test for that scenario, but seems like a good thing to have.

If `skip_locked=True` is problematic with `.get(...)`, maybe try `nowait=True` and catch `DatabaseError`?

Reply:
I did write up this test locally, and as you guessed, a `DoesNotExist` error was raised when `skip_locked=True`. If we changed that to `nowait=True`, an `OperationalError` was raised. I've added a test to hit this scenario in the alternative PR I've created now that we are no longer handling exceptions. (https://github.com/dimagi/commcare-hq/pull/33745)